### ISSUE TITLE
Spec: Test email delivery

### ISSUE THREAD

**Issue by dottorer** [lale-help/lale-help#275]
We need to test that the Mandrill integration and email delivery always works. Therefore we should send an email to 'info@lale.help' using a template and some merge variables.

Per: https://docs.google.com/spreadsheets/d/1EKSlUwtFJV-1yHM3jx2-LKpE1TsOn43udqrJZodvmD4/edit#gid=0

**Comment by pjfanning** [comment]
Are we looking to have the deployed app (lale.help, staging.lale.help, etc.) send periodic emails to a set email address to verify Mandrill/MailChimp connectivity? Or are we looking for rspec tests that test the mandrill templates?

**Comment by dottorer** [comment]
@pjfanning I think we need to test that emails are going out in general for the different envs. The entire suite of templates will be something we aspire to, but for now it would be good if we can verify with a test email using some bind variables that email do get delivered. This would be for environments:

app.lale.help ("PROD")
staging.lale.help ("DEV")
demo.lale.help ("DEMO")

Thank you!!!

**Comment by pjfanning** [comment]
Thanks Ralf for clarifying the requirement. One approach would be to set up a scheduled task on the 3 deployments and to have it send an email on a regular basis. We'd have to find some way of checking the mailbox to see if the emails are arriving. This would be easy enough to automate, the main question would be how to report issues
We could also catch any exceptions that happen on the Mandrill API and maybe set up some Rollbar integration that fetches and reports such exceptions. Not all delivery failures would be caused by Mandrill API failures but it could be useful to have this 2nd form of monitoring too. This approach could be used to also monitor status of services like the database, etc.

**Comment by dottorer** [comment]
Yes, I was wondering on how we could verify the email delivery. We could setup a dedicated email ("testing@lale.help") and route that to me as an alias. This way I could verify manually that email were sent as part of the build. Better is of course to detect this and throw a rollbar error. I am sure this has been done tons of time, maybe @phillipoertel or @phil-monroe know a good way.

**Comment by phillipoertel** [comment]
Hi guys, 

My two cents (without reading your full thread): don't cross the application boundaries in these tests. Write tests to ensure the emails leave Rails where we expect them to, and trust the professional providers (Mandrill, SMTP servers) to get the rest of the job done. 
I believe email integration tests would be complex to set up, verify and maintain, and provide little additional value to what I describe above. 

Cheers Phillip 

Von unterwegs gesendet, deswegen kurz und evtl. mit Tippfehlern

**Comment by pjfanning** [comment]
I see some merit in trusting MailChimp to deliver the emails. Ralf, does MailChimp app provide statistics about emails that have been sent and/or tracking of issues with API requests?
The idea of catching Mandrill API exceptions in the Lale app and monitoring them still seems valid if MailChimp doesn't do this for us.

**Comment by dottorer** [comment]
Mailchimp definitely has statistics and successful delivery of emails (vs. failed due to wrong address, etc.). They also track the click rate by the redirection URL in the emails. But mostly we are trying to ensure that our primary comms channel of email is robustly working after a checking / build.
